python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -705,6 +707,114 @@ def cummax(self):
         """Get the column-wise cumulative maximum value in each group."""
         return self.agg("cummax")
 
+    def _scan_fill(self, method: str, limit: int) -> DataFrameOrSeries:
+        """Internal implementation for `ffill` and `bfill`
+        """
+        value_columns = self.grouping.values()
+        result = self._groupby.replace_nulls(
+            Table(value_columns._data), method
+        )
+        result = self.obj.__class__._from_table(result)
+        result = self._mimic_pandas_order(result)
+        return result._copy_type_metadata(value_columns)
+
+    def pad(self, limit=None):
+        """Forward fill NA values.
+
+        Parameters
+        ----------
+        limit : int, default None
+            Unsupported
+        """
+
+        if limit is not None:
+            raise NotImplementedError("Does not support limit param yet.")
+
+        return self._scan_fill("ffill", limit)
+
+    ffill = pad
+
+    def backfill(self, limit=None):
+        """Backward fill NA values.
+
+        Parameters
+        ----------
+        limit : int, default None
+            Unsupported
+        """
+        if limit is not None:
+            raise NotImplementedError("Does not support limit param yet.")
+
+        return self._scan_fill("bfill", limit)
+
+    bfill = backfill
+
+    def fillna(
```

Comment:
Should we also provide an option whether we need to follow pandas or not, and by default follow it ?

Reply:
Previous discussion with @shwina mentions that we probably don't want to introduce too many parameters to further confuse users. Especially at `groupby(sort=False)` level we already differ from Pandas.

Let that be said, `sort=False` doesn't seem to be oberseved by pandas for transform-like operation either:
```
>>> df = pd.DataFrame({'a': [3, 2, 3, 2], 'b': [5, None, None, 8]})
>>> df.groupby('a', sort=True).fillna(method='ffill')
     b
0  5.0
1  NaN
2  5.0
3  8.0
>>> df.groupby('a').fillna(method='ffill')
     b
0  5.0
1  NaN
2  5.0
3  8.0
```

cc @shwina